src/core/Fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/fieldset.css';
+
+export type FieldsetProps = {
+  /**
+   * The caption or title for the fieldset
+   */
+  legend?: React.ReactNode;
+} & React.ComponentPropsWithoutRef<'fieldset'>;
+
+/**
+ * Basic fieldset component to group different components together
```

Comment:
Maybe update the wording here and add period at the end too.

```suggestion
 * Fieldset component to group several inputs, controls and labels within a form.
```